CMakeLists.txt

```diff
@@ -338,7 +339,9 @@ find_package(OpenSSL QUIET)
 # ==============================================================================
 add_subdirectory(cmake)
 add_subdirectory(include)
-add_subdirectory(lib)
+if (NOT CUDAQ_DISABLE_FRONTEND)  
```

Comment:
First attempt at allowing someone to disable the frontend. Could use input on name preferences for this flag, and if there are any other lines in the file that should be wrapped up in this conditional

Reply:
You'll need to go a bit more fine-grained into this lib folder, specifically we'll want the Optimizer, Support, and Target folders to build, just not Frontend.